torch/quantization/observer.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch
+from functools import partial
+
+class Observer(nn.Module):
+    r"""Default Observer Module
+    A default implementation of the observer module, only works for
+    `per_tensor_affine` quantization scheme.
+    The module will record the running average of max and min value of the
+    observed Tensor and calulate_qparams will calculate the scale and zero_point
+
+    Other types of Observers should follow the same API, it can take arbitrary
+    number of keyward arguments. In forward, it will update the statistics of
+    the observed Tensor. And it should provide a `calculate_qparam` function
+    that computes the quantization parameters given the collected statistics.
+    TODO: Maybe add an abstract Observer class that enforces these rules?
+    """
+    def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine, avg_constant=0.9):
+        super(Observer, self).__init__()
+        self.dtype = dtype
+        self.qscheme = qscheme
+        assert self.qscheme == torch.per_tensor_affine, \
+            'Default Observer only works for per_tensor_affine quantization scheme'
+        # Symmetric range for initialization
+        self.stats = torch.tensor([-6, 6], dtype=torch.float)
```

Comment:
btw, why are we doing exponential smoothing? I'd imagine explicitly calculating some percentile of distribution would be much more meaningful default. The problem with exponential smoothing is that it biases towards last batches and also -6/6 doesn't make sense.

@raghuramank100 ?

Reply:
Agreed, lets go with min/max , just like we do for weights.